code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm

```diff
@@ -0,0 +1,412 @@
+/datum/reagent/inverse/healing/tirimol/on_mob_life(mob/living/carbon/owner, delta_time, times_fired)
+	switch(current_cycle)
+		if(1 to 10)//same delay as chloral hydrate
+			if(prob(50))
+				owner.emote("yawn")
+		if(10 to INFINITY)
+			owner.Sleeping(40)
+			. = 1
+			if(owner.IsSleeping() && !cached_reagent_list)
+				for(var/datum/reagent/reagent as anything in owner.reagents.reagent_list)
+					if(reagent in cached_reagent_list)
+						continue
+					if(istype(reagent, /datum/reagent/medicine))
+						if(!(reagent.creation_purity > reagent.inverse_chem_val))//Only affect pure types
+							continue
+						reagent.creation_purity *= 1.25
+						cached_reagent_list += reagent
+
+			else if(!owner.IsSleeping() && cached_reagent_list)
+				for(var/datum/reagent/reagent as anything in cached_reagent_list)
+					if(!reagent)
+						continue
+					reagent.creation_purity *= 0.8
+				cached_reagent_list = list()
+	..()
+
+/datum/reagent/inverse/healing/tirimol/on_mob_delete(mob/living/owner)
+	if(owner.IsSleeping())
+		owner.visible_message("<span class='notice'>[icon2html(owner, viewers(DEFAULT_MESSAGE_RANGE, src))] [owner] lets out a hearty snore!</span>")//small way of letting people know the supersnooze is ended
+	for(var/datum/reagent/reagent as anything in cached_reagent_list)
+		if(!reagent)
+			continue
+		reagent.creation_purity *= 0.8
+	cached_reagent_list = list()
+	..()
+
+//seiver
+////Inverse
+//Allows the scanner to detect organ health to the nearest 1% (similar use to irl) and upgrates the scan to advanced
+/datum/reagent/inverse/technetium
+	name = "Technetium 99"
+	description = "A radioactive tracer agent that can improve a scanner's ability to detect internal organ damage. Has a very low metabolism rate and will irradiate the patient when present, purging is recommended after use."
+	metabolization_rate = 0.01 * REM
+	chemical_flags = REAGENT_DONOTSPLIT //Do show this on scanner
+	tox_damage = 0
+
+/datum/reagent/inverse/technetium/on_mob_life(mob/living/carbon/owner, delta_time, times_fired)
+	owner.radiation += creation_purity * delta_time // 0 - 1
+
+//Kind of a healing effect, Presumably you're using syrinver to purge so this helps that
+/datum/reagent/inverse/healing/syriniver
+	name = "Syrinifergus"
+	description = "This reagent reduces the impurity of all non medicines within the patient, reducing their negative effects."
+	///The list of reagents we've affected
+	var/cached_reagent_list = list()
+	addiction_types = list(/datum/addiction/medicine = 1.75)
+
+/datum/reagent/inverse/healing/syriniver/on_mob_add(mob/living/living_mob)
+	if(!(iscarbon(living_mob)))
+		return ..()
+	var/mob/living/carbon/owner = living_mob
+	for(var/datum/reagent/reagent as anything in owner.reagents.reagent_list)
+		if(!(istype(reagent, /datum/reagent/medicine)))
+			continue
+		if(!(reagent.creation_purity > reagent.inverse_chem_val))//Only affect pure types
+			continue
```

Comment:
there's no check to see if a reagent is already in the cached_reagent_list here

Reply:
oops, fixed.